cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -144,12 +145,21 @@ func MustNew(cfg *config.Config) *Daemon {
 	if len(cfg.HistoryArchiveURLs) == 0 {
 		logger.Fatal("no history archives url were provided")
 	}
-	historyArchive, err := historyarchive.Connect(
-		cfg.HistoryArchiveURLs[0],
+
+	historyArchive, err := historyarchive.NewArchivePool(
+		cfg.HistoryArchiveURLs,
 		historyarchive.ArchiveOptions{
+			NetworkPassphrase:   cfg.NetworkPassphrase,
 			CheckpointFrequency: cfg.CheckpointFrequency,
+			ConnectOptions: storage.ConnectOptions{
+				Context:   context.Background(),
+				UserAgent: cfg.HistoryArchiveUserAgent},
+			CacheConfig: historyarchive.CacheOptions{
```

Comment:
now that we're setting `Cache` to false, you can omit the `CacheConfig` property entirely

Reply:
just for future reference, this will need a minor update once we back-merge 2.28.2 which will just be changing the cache path to be empty

Reply:
yes, i had emitted it locally and then the visual of disabled here seemed worth it, but np removing as well - https://github.com/stellar/soroban-rpc/pull/56/commits/0600fe94542e395191065eb2ca51550f2c66ff65